code/script.js

```diff
@@ -222,104 +227,189 @@ const generateBoard = () => {
 }
 
-// This function to start (and restart) the game
+// Function: starts the game. 
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  charactersInPlay = CHARACTERS // Here we're setting charactersInPlay array to be all the characters to start with
 }
 
-// setting the currentQuestion object when you select something in the dropdown
+// Function: selectQuestion sets the currentQuestion when the user selects something in the dropdown (ie when the user formulates his/her question).
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
+  console.log('This is the question that I want answered')
+  const category = questions.options[questions.selectedIndex].parentNode.label // This variable stores what option group (category) the question belongs to.
+  const value = questions.value // This variable stores the actual value of the quesiton we've selected in the dropdown.
+    
+    if (category === 'hair') {
+      console.log('hair:', value)
+      currentQuestion = { 
+        category: category,
+        value: value
+      }
+    }
+  
+    else if (category === 'eyes') {
+      console.log('eyes:', value)
+      currentQuestion = { 
+        category: category,
+        value: value
+      }
+    }
 
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+    else if (category === 'accessories') {
+      console.log('accessories:', value)
+      currentQuestion = { 
+        category: category,
+        value: value 
+      }
+    }
 
-  currentQuestion = {
-    category: category,
-    // value: value
+    else if (category === 'other') {
+      console.log('other:', value)
+      currentQuestion = {
+        category: category,
+        value: value 
+      }
+     }
   }
-}
 
-// This function should be invoked when you click on 'Find Out' button.
+// Function: checkQuestion is invoked when pressing the findOutButton, and leads to the filter function. 
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  console.log('Please check if the secret person has these attributes!')
+  const { category, value } = currentQuestion 
+  
+  let keep
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
   if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
-
+    console.log(category, value) // Daniels förslag 
+    if (value === secret.hair || value === secret.eyes) {
+      keep = true
```

Comment:
Since an object's property's value can be accessed with square brackets, this could be shortened as follows:
`if (value === secret[category])` to avoid having to compare the hair and eyes separately (category will be storing either 'hair' or 'eyes', depending on what the player has selected). 
Same goes for row 303. 

I also don't think you need to declare `keep` at all - either as a global variable above or in this function - since it is just a placeholder for `filterCharacters()`. 
I think once you call `filterCharacters()` the computer will check if there is or isn't anything inside the parenthesis and then follows the code accordingly..(this is my unprofessional explanation without correct terminology, but it works in my code that's why I know :D )

Reply:
Great, thank you!